File: src/main/java/ru/protei/oxmldoc/common/Row.java
package ru.protei.oxmldoc.common;

import ru.protei.oxmldoc.style.CellStyle;
import ru.protei.oxmldoc.style.RowStyle;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.function.Supplier;

public class Row {
    private final List<Cell<?>> cells = new ArrayList<>();
    private RowStyle rowStyle;

    public Row(RowStyle style) {
        this.rowStyle = style;
    }

    public Row() {
    }

    public List<Cell<?>> getCells() {
        return cells;
    }

    public RowStyle getRowStyle() {
        return rowStyle;
    }

    public void setRowStyle(RowStyle rowStyle) {
        this.rowStyle = rowStyle;
    }

    public Row append(Cell<?> cell) {
        cells.add(cell);
        return this;
    }

    public Row append(String x) {
        cells.add(new Cell<>(x, null));
        return this;
    }

    public Row append(String x, CellStyle s) {
        cells.add(new Cell<>(x, s));
        return this;
    }

    public Row append(Date x) {
        cells.add(new Cell<>(x, null));
        return this;
    }

    public Row append(Date x, CellStyle s) {
        cells.add(new Cell<>(x, s));
        return this;
    }

    public Row append(Number x) {
        cells.add(new Cell<>(x, null));
        return this;
    }

    public Row append(Number x, CellStyle s) {
        cells.add(new Cell<>(x, s));
        return this;
    }

    public Row append(Supplier<String> func) {
        cells.add(new Cell<>(func, null));
        return this;
    }

    public Row append(Supplier<String> func, CellStyle s) {
        cells.add(new Cell<>(func, s));
        return this;
    }

    public Row append() {
        cells.add(new Cell<Void>(null, null));
        return this;
    }

    public void clear() {
        cells.clear();
    }
}

File: src/main/kotlin/ru/protei/oxmldoc/common/Row.kt
package ru.protei.oxmldoc.common

import ru.protei.oxmldoc.style.CellStyle
import ru.protei.oxmldoc.style.RowStyle
import java.util.*
import kotlin.collections.ArrayList

class Row(val rowStyle: RowStyle?, val cells: MutableList<Cell<*>>) {
    constructor(): this(null, ArrayList<Cell<*>>())

    constructor(style: RowStyle) : this(style, ArrayList<Cell<*>>())

    fun append(cell: Cell<*>) {
        cells.add(cell)
    }

    fun append(x: String) {
        cells.add(Cell(x, null))
    }

    fun append(x: String, s: CellStyle) {
        cells.add(Cell(x, s))
    }

    fun append(x: Date): Unit {
        cells.add(Cell(x, null))
    }

    fun append(x: Date, s: CellStyle) {
        cells.add(Cell(x, s))
    }

    fun append(x: Number) {
        cells.add(Cell(x, null))
    }

    fun append(x: Number, s: CellStyle) {
        cells.add(Cell<Number>(x, s))
    }

    fun append(f: (String) -> String) {
        cells.add(Cell(f, null))
    }

    fun append(f: (String) -> String, s: CellStyle) {
        cells.add(Cell(f, s))
    }

    fun append() {
        cells.add(Cell(null, null))
    }

    fun clear() {
        cells.clear()
    }
}